### ISSUE TITLE
bring up QEMU cluster with GoVM

### ISSUE THREAD

**Issue by pohly** [intel/pmem-CSI#193]
https://github.com/govm-project/govm looks like a good replacement for the home-grown scripts that configures the QEMU cluster and bring up Clear Linux in it. By switching to GoVM, we make the process simpler for users (automatic configuration of network) and more flexible (can support other OSes).

**Comment by okartau** [comment]
It is flexible enough to support everything we use now? I am firstly and mostly concerned about added qemu options for nvdimm emulation.

**Comment by pohly** [comment]
The internal CI uses it, so it should be good enough. It has a config file where one can directly add more QEMU command line options.

**Comment by okartau** [comment]
I made a script set that makes almost same with govm and Ubuntu than `make start` does with Clear. The cluster gets deployed and pmem-csi pods get to Running state, thats good.
I also tried to keep common part separate so that adding next distro will be easier.
PR coming soon!

The downside is,  pmem device side does not work yet.
Ubuntu kernel in those cloud-provided images that govm uses:
https://cloud-images.ubuntu.com/
seems not to support pmem devices same way as is needed to recognize and use pmem block devices.
I tried 18.04 and 18.10 images, same story: `pfn: failed to enable`, etc messages, and I never see /dev/pmem* devices. 
I don't see nd_pmem under /sys/bus/nd/drivers/ :
```
root@host-2:~# ls -l /sys/bus/nd/drivers/ 
drwxr-xr-x 2 root root 0 Apr  9 16:56 nd_bus
drwxr-xr-x 2 root root 0 Apr  9 16:56 nd_region
drwxr-xr-x 2 root root 0 Apr  9 16:56 nvdimm
```

When I manually install different regular kernel, one of those at, for example
http://kernel.ubuntu.com/~kernel-ppa/mainline/v4.19.13/
(this is known to work as I use it on a VM for pmem-csi development)
then I get nvdimm devices (emulated) appear as they should.
drivers list has nd_pmem:
```
root@host-1:~# ls -l /sys/bus/nd/drivers/ 
drwxr-xr-x 2 root root 0 Apr  9 19:56 dax_pmem
drwxr-xr-x 2 root root 0 Apr  9 19:56 nd_bus
drwxr-xr-x 2 root root 0 Apr  9 19:56 nd_pmem
drwxr-xr-x 2 root root 0 Apr  9 19:56 nd_region
drwxr-xr-x 2 root root 0 Apr  9 19:56 nvdimm
```

Either we have to make VM setup more complex by replacing kernel,
or we can continue searching for different "cloud" flavor images.

**Comment by okartau** [comment]
I started change set #238 to make tester system more generic so that it becomes easier to implement different distros-based testers. Naming changes are there already. Another thing that needs changes is hard-coded IP addresses of testing code. Clear-based VMs get address 192.168.x.x but govm-Ubuntu based VMs get address 172.17, so we should make this dynamically specified.

**Comment by pohly** [comment]
Resolved by PR #236
